src/status_im/ui2/screens/chat/composer/input.cljs

```diff
@@ -168,12 +174,130 @@
       :on-selection-change      (partial on-selection-change timeout-id last-text-change mentionable-users)
       :on-change                (partial on-change last-text-change timeout-id mentionable-users refs chat-id sending-image)
       :on-text-input            (partial on-text-input mentionable-users chat-id)}
-     (if mentions-enabled
-       (for [[idx [type text]] (map-indexed
-                                (fn [idx item]
-                                  [idx item])
-                                (<sub [:chat/input-with-mentions]))]
-         ^{:key (str idx "_" type "_" text)}
-         [rn/text (when (= type :mention) {:style {:color colors/primary-50}})
-          text])
-       (get @input-texts chat-id))]))
+     (fn []
+       (if mentions-enabled
+         (for [[idx [type text]] (map-indexed
+                                  (fn [idx item]
+                                    [idx item])
+                                  (<sub [:chat/input-with-mentions]))]
+           ^{:key (str idx "_" type "_" text)}
+           [rn/text (when (= type :mention) {:style {:color colors/primary-50}})
+            text])
+         (get @input-texts chat-id)))]))
+
+(defn selectable-text-input-manager []
+  (when (exists? (.-NativeModules react-native))
+    (.-RNSelectableTextInputManager ^js (.-NativeModules react-native))))
+
+(defonce rn-selectable-text-input (reagent/adapt-react-class (.requireNativeComponent react-native "RNSelectableTextInput")))
+
+(declare first-level-menu-items second-level-menu-items)
+
+(defn update-input-text [{:keys [text-input chat-id]} text]
+  (on-text-change text chat-id)
+  (.setNativeProps ^js text-input (clj->js {:text text})))
+
+(defn calculate-input-text [{:keys [full-text selection-start selection-end]} content]
+  (let [head (subs full-text 0 selection-start)
+        tail (subs full-text selection-end)]
+    (str head content tail)))
+
+(def first-level-menus {:cut (fn [{:keys [content] :as params}]
+                               (let [new-text (calculate-input-text params "")]
+                                 (react/copy-to-clipboard content)
+                                 (update-input-text params new-text)))
+
+                        :copy-to-clipboard (fn [{:keys [content]}]
+                                             (react/copy-to-clipboard content))
+
+                        :paste (fn [params]
```

Comment:
Since all operations happens on the clipboard, Can we rename the key `:copy-to-clipboard` to `:copy``

Reply:
i don't think so, as we may use this keyword relate to i18n